### PR TITLE
Php2branch >tl1revbranch

### PR DESCRIPTION
**Pull request by php-2**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php [hunk 1]
```diff
@@ -3309,7 +3309,7 @@ function formAddObjectLine($dateSelector,$seller,$buyer)
 		require_once DOL_DOCUMENT_ROOT.'/core/class/extrafields.class.php';
 		$extrafieldsline = new ExtraFields($this->db);
 		$extralabelslines=$extrafieldsline->fetch_name_optionals_label($this->table_element_line);
-
+
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php]: Why you doing changes here?

File: portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php [hunk 2]
```diff
@@ -3431,8 +3431,7 @@ function printObjectLines($action, $seller, $buyer, $selected=0, $dateSelector=0
     		print '<td class="linecoldelete" width="10"></td>';
 
     		print '<td class="linecolmove" width="10"></td>';
-
-    		print "</tr>\n";
+                 		print "</tr>\n";
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php]: Why you doing changes here?

File: portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php [hunk 3]
```diff
@@ -174,6 +174,23 @@ function addExtraField($attrname, $label, $type, $pos, $size, $elementtype, $uni
 			return -1;
 		}
 	}
+        //new code added by kamal.asran
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php]: you MUST mention date at your comment.

File: portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php [hunk 4]
```diff
@@ -174,6 +174,23 @@ function addExtraField($attrname, $label, $type, $pos, $size, $elementtype, $uni
...
+        function checkProduct($ref,$label,$sales_tax,$up_net,$up_currency,$qty,$reduce,$cost_price,$total_net,$total_net_in_currency)
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php]: Why you put this function at this file!!!

File: portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php (hunk 1 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php]: i just tested some thing and forget to back it again ,but now i back up to its previous version

File: portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php (hunk 2 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/core/class/commonobject.class.php]: i backed  it up to its previous version

File: portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php (hunk 3 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php]: ok

File: portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php (hunk 4 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/core/class/extrafields.class.php]: just i want to use any file that contain function that deals with database

File: portfolio/ifintech/v1/ierp/htdocs/comm/propal/card.php [hunk 5]
```diff
@@ -2188,31 +2189,106 @@
...
+                include 'excel_reader.php';     // include the class
+                mb_internal_encoding("8bit");
+                $excel = new PhpExcelReader;
+                $excel->read('iERP-Migration.xls');
+                $num_products = $excel->sheets[0]['numRows'];
+                $num_products--;
+                $prop_id=GETPOST("id");
+                $error_massage='';
+                $sql='';
+                $sql .="INSERT INTO ".MAIN_DB_PREFIX."propaldet (";
```
[the remark this answers is not in the document]
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/comm/propal/card.php]: i want to insert multi rows in one sql statement so i defined  this statement first ant then concat it 
every sql statement in the loop

File: portfolio/ifintech/v1/ierp/htdocs/comm/propal/class/propal.class.php [hunk 6]
```diff
@@ -586,6 +586,7 @@ function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1=0.0, $txlocaltax2=0.0
             else
             {
                 $this->error=$this->line->error;
+                $this->error;
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/comm/propal/class/propal.class.php]: what this line for ?
$this->error;

File: portfolio/ifintech/v1/ierp/htdocs/user/index.php [hunk 7]
```diff
@@ -674,7 +674,46 @@
...
+            {
+                include 'excel_reader.php';     // include the class
+                mb_internal_encoding("8bit");
+                $excel = new PhpExcelReader;
+                $excel->read('iERP-Migration.xls');
+                $num_users = $excel->sheets[3]['numRows'];
+                $num_users--;
+                $error_massage='';
+                $sql='';
+                $sql .="INSERT INTO ".MAIN_DB_PREFIX."user (";
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/user/index.php]: we should not use sql to insert records!
we should use the existing code already with the validations.

File: portfolio/ifintech/v1/ierp/htdocs/comm/propal/class/propal.class.php (hunk 6 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/comm/propal/class/propal.class.php]: i really sorry , i wrote that statement in echo $this->error to test if error occurs inside create function but deleted echo and forget to delete $this->error;
i deleted this line now

File: portfolio/ifintech/v1/ierp/htdocs/user/index.php (hunk 7 above)
Comment (php-2) [reply portfolio/ifintech/v1/ierp/htdocs/user/index.php]: it's the old code i wrote it before using existing code in dolibarr , i used the existing code and method in dolibarr in product items and products and services but not did that for users till now,
i will delete it and use the code existing in dolibarr

**Pull request merged**